typing/jkind.ml

```diff
@@ -793,11 +793,12 @@ module Const = struct
   let get_required_layouts_level (_context : History.annotation_context)
       (jkind : t) : Language_extension.maturity =
     match jkind.layout, jkind.nullability_upper_bound with
-    | (Base (Float64 | Float32 | Word | Bits32 | Bits64 | Vec128) | Any), _
+    | ( ( Base (Float64 | Float32 | Word | Bits32 | Bits64 | Vec128)
+        | Any | Product _ ),
```

Comment:
Does this need to recur into `Product`?

Reply:
Good catch - I've also added a test now.